src/kudu/consensus/log_cache.cc

```diff
@@ -225,22 +423,59 @@ Status LogCache::AppendOperations(const vector<ReplicateRefPtr>& msgs,
   l.unlock();
 
   metrics_.log_cache_size->IncrementBy(mem_required);
+  metrics_.log_cache_msg_size->IncrementBy(total_msg_size);
   metrics_.log_cache_num_ops->IncrementBy(msgs.size());
 
-  Status log_status = log_->AsyncAppendReplicates(
-    msgs, Bind(&LogCache::LogCallback,
-               Unretained(this),
-               last_idx_in_batch,
-               borrowed_memory,
-               callback));
+  Status log_status;
+  if (!found_compressed_msgs) {
+    // No compressed messages are found, so appen orinals msgs to log
+    log_status = log_->AsyncAppendReplicates(
+      msgs, Bind(&LogCache::LogCallback,
+                 Unretained(this),
+                 last_idx_in_batch,
+                 borrowed_memory,
+                 callback));
+  } else {
+    // The batch contains some compressed msgs. It needs to be uncompressed
+    // before appending to the log
+    vector<ReplicateRefPtr> uncompressed_msgs;
+    uncompressed_msgs.reserve(msgs.size());
+
+    for (const auto& msg : msgs) {
+      const auto compression_codec =
+        msg->get()->write_payload().compression_codec();
+
+      if (compression_codec == NO_COMPRESSION) {
+        // msg is not compressed, use it to write to the log
+        uncompressed_msgs.push_back(msg);
+      } else {
+        // msg needs to be uncompressed before writing to the log
+        std::unique_ptr<ReplicateMsg> uncompressed_msg;
+        auto status = UncompressMsg(
+            msg, log_cache_compression_buf_, &uncompressed_msg);
+
+        // Crash if uncompression failed
+        CHECK_OK_PREPEND(status,
```

Comment:
if we have compressed messages we are a FOLLOWER/LEARNER, Instead of crashing can we reject these messages before StartFollowerTransaction

Reply:
Is this fails, then Append() fails and SFT crashes anyways - I believe it is beyond point of no return